core/src/main/java/io/questdb/cairo/TableNameRegistryRW.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo;
+
+import io.questdb.cairo.wal.AbstractTableNameRegistry;
+import io.questdb.std.Chars;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+public class TableNameRegistryRW extends AbstractTableNameRegistry {
+    private final ConcurrentHashMap<CharSequence, TableToken> nameTableTokenMap = new ConcurrentHashMap<>();
+    private final ConcurrentHashMap<TableToken, String> reverseTableNameTokenMap = new ConcurrentHashMap<>();
```

Comment:
we should use our `ConcurrentHashMap` keyed on directory name, while using `TableToken` in the API of the registry

Reply:
ok, done